src/sources/util/tcp.rs

```diff
@@ -115,17 +117,47 @@ pub trait TcpSource: Clone + Send + 'static {
 
                         let out = out.clone();
 
-                        let events_in = FramedRead::new(socket, source.decoder())
-                            .take_until(tripwire)
-                            .filter_map(move |frame| {
-                                let host = host.clone();
-                                source.build_event(frame, host)
-                            })
-                            .map_err(|error| warn!(message = "connection error.", %error));
-
-                        let handler = events_in.forward(out).map(|_| debug!("connection closed."));
-
-                        tokio::spawn(handler.instrument(span.clone()));
+                        if let Some(tls) = tls.clone() {
+                            match tls.acceptor() {
+                                Err(error) => error!(message = "Failed to create a TLS connection acceptor", %error),
+                                Ok(acceptor)=> {
```

Comment:
can we break out this nested iflet matches a bit so that its a bit easier to follow?

Reply:
I pushed up a030d995b31a9f5a82f2ca82c67ad9324bd8473c, is that what you were suggesting?